engine/src/main/java/org/terasology/rendering/dag/DirectionalLightsNode.java

```diff
@@ -0,0 +1,149 @@
+import static org.lwjgl.opengl.GL11.glEnable;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+
+/**
+ * TODO: Diagram of this node
+ */
+public class DirectionalLightsNode implements Node {
+
+    @In
+    private BackdropProvider backdropProvider;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
+    // TODO: Review this? (What are we doing with a component not attached to an entity?)
+    private LightComponent mainDirectionalLight = new LightComponent();
+
+    private Camera playerCamera;
+    private FBO sceneOpaque;
+    private Material lightGeometryShader;
+    private Material lightBufferPass;
+
+    @Override
+    public void initialise() {
+        playerCamera = worldRenderer.getActiveCamera();
+        sceneOpaque = frameBuffersManager.getFBO("sceneOpaque");
+        lightGeometryShader = worldRenderer.getMaterial("engine:prog.lightGeometryPass");
+        lightBufferPass = worldRenderer.getMaterial("engine:prog.lightBufferPass");
+        initMainDirectionalLight();
+    }
+
+    // TODO: one day the main light (sun/moon) should be just another light in the scene.
+    private void initMainDirectionalLight() {
+        mainDirectionalLight.lightType = LightComponent.LightType.DIRECTIONAL;
+        mainDirectionalLight.lightAmbientIntensity = 0.75f;
+        mainDirectionalLight.lightDiffuseIntensity = 0.75f;
+        mainDirectionalLight.lightSpecularPower = 100f;
+    }
+
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/directionallights");
+        sceneOpaque.bind();
+
+        Vector3f sunlightWorldPosition = new Vector3f(backdropProvider.getSunDirection(true));
+        sunlightWorldPosition.scale(50000f);
+        sunlightWorldPosition.add(playerCamera.getPosition());
+        // TODO: find a more elegant way
+        // TODO: iterating over RenderSystems for rendering multiple lights
+        worldRenderer.renderLightComponent(mainDirectionalLight, sunlightWorldPosition, lightGeometryShader, false);
+
+        glDisable(GL_BLEND);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+
+        glEnable(GL_DEPTH_TEST);
+
+        setRenderBufferMask(sceneOpaque, true, true, true);
+        bindDisplay();
+
+        applyLightBufferPass();
+        PerformanceMonitor.endActivity();
+    }
+
+    /**
+     * Part of the deferred lighting technique, this method applies lighting through screen-space
+     * calculations to the previously flat-lit world rendering stored in the primary FBO.   // TODO: rename sceneOpaque* FBOs to primaryA/B
+     * <p>
+     * See http://en.wikipedia.org/wiki/Deferred_shading as a starting point.
+     */
+    private void applyLightBufferPass() {
+        int texId = 0;
+
+        GL13.glActiveTexture(GL13.GL_TEXTURE0 + texId);
+        sceneOpaque.bindTexture();
+        lightBufferPass.setInt("texSceneOpaque", texId++, true);
```

Comment:
These three lines blocks are used quite often, either here or in the ShaderParameters classes, we might want to invent methods that wrap these into more compact code. Not needed yet though.